### ISSUE TITLE
Claude-generated: Wrong gradients through PSD cone slacks when equilibration is on

### ISSUE THREAD

**Issue by sbarratt** [moreau-project/moreau#58]
Claude-generated. Found by an external review of 0.4.1 and re-verified on main @ e794100 (CPU).

## Problem

With equilibration enabled (the default), gradients through the slacks of **inactive** PSD cones are wrong. At an optimum where both PSD constraints are inactive (`z = 0`), `x* = −P⁻¹q` does not depend on `b`, so `s = b − Ax*` gives `d(sum s)/db = 1` exactly. Moreau returns 0.77 to 1.09 per entry; with `equilibrate_enable=False` it is exact. The backward pass does not consistently undo the (cone-rectified) equilibration scaling for PSD blocks. The reviewer also saw ~1e-3 errors for inactive exp/SOC blocks.

## Reproducer

```python
import numpy as np, torch, moreau
from moreau.torch import Solver
d = dict(
    n=2, m=9,
    P_ro=[0, 2, 4], P_ci=[0, 1, 0, 1],
    P_values=[0.6256723609278945, -0.0508244538467591, -0.0508244538467591, 0.46705182220947905],
    A_ro=[0, 1, 2, 3, 4, 6, 7, 8, 9, 10], A_ci=[1, 0, 1, 1, 0, 1, 0, 0, 1, 0],
    A_values=[-0.4837139692280515, -0.06020557278290741, -0.5840219388984813, -0.004622389939694887,
              1.3221019133427179, 0.4141595421316479, 1.3008229359218288, 0.515627331611208,
              0.26175515511212616, 1.1873828775582242],
    q=[-0.2984003384219467, -0.11898003687945923],
    b=[0.4381098646925604, 0.01529467393027768, 0.38170275423656597, 0.10298275007370732,
       0.8525821243992978, 1.2571806058658785, 0.7747668064388958, 0.0578480607000114, 1.229085694225883],
    cones=dict(psd_dims=[3, 2]),
)
f64 = lambda a: torch.tensor(a, dtype=torch.float64)
for equil in (True, False):
    st = moreau.Settings(device="cpu", solver="ipm", ipm_settings=moreau.IPMSettings(equilibrate_enable=equil))
    t = torch.tensor
    s = Solver(d["n"], d["m"], t(d["P_ro"]), t(d["P_ci"]), t(d["A_ro"]), t(d["A_ci"]),
               moreau.Cones(**d["cones"]), settings=st)
    b = f64(d["b"]).requires_grad_()
    r = s.solve(f64(d["P_values"]), f64(d["A_values"]), f64(d["q"]), b)
    (r.s.sum() + r.z.sum()).backward()
    print(equil, np.round(b.grad.numpy(), 4))   # expected all 1.0
```

```
True  [1.088  1.0126 1.0952 1.0082 0.7657 0.8473 0.9999 0.9999 0.9995]
False [1. 1. 1. 1. 1. 1. 1. 1. 1.]
```

## Scope

- [ ] Fix the PSD-block unscaling in the backward pass when equilibration is on; check exp/SOC blocks for the same ~1e-3 discrepancy.
- [ ] Regression test with this problem (and a CUDA run once #34 lands).
